### PR TITLE
Additional Psycopg tests

### PR DESCRIPTION
**Pull request by elpavel**
##### SUMMARY
Modified existing tests to work with Psycopg 3 and added extra Psycopg related tests.

The postgresql_user module handles the "database is read only" error, although only when altering a user. Tests were added to verify the behaviour.

Tests were added to verify the path in postgresql_query when `encoding` is specified.

`query_result` should be a list. When the list is empty, currently using psycopg2 we return an empty dict. When psycopg3 is used the same code will return an empty list. I think it is OK to keep the current behaviour, but no need to preserve that for psycopg3.

##### ISSUE TYPE
- Additional tests

##### COMPONENT NAME
Test harness

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: tests/integration/targets/postgresql_query/tasks/postgresql_query_initial.yml [hunk 1]
```diff
@@ -41,8 +41,8 @@
       - result.query_list == ['ANALYZE test_table']
       - result.rowcount == 0
       - result.statusmessage == 'ANALYZE'
-      - result.query_result == {}
-      - result.query_all_results == [{}]
+      - result.query_result == {} or result.query_result == []  # Psycopg2 or Psycopg 3
+      - result.query_all_results == [{}] or result.query_all_results == [[]]  # Psycopg2 or Psycopg 3
```
Comment (Andersson007) [inline tests/integration/targets/postgresql_query/tasks/postgresql_query_initial.yml:45]: Do you think if it's impossible to convert returned rows to dictionaries in Psycopg 3?
```
query_result:
    description:
    - List of dictionaries in column:value form representing returned rows.
    - When running queries from a file, returns result of the last query.
    returned: success
    type: list
    elements: dict
    sample: [{"Column": "Value1"},{"Column": "Value2"}]
```

Comment (Andersson007) [reply tests/integration/targets/postgresql_query/tasks/postgresql_query_initial.yml:45]: asking as it'd be a breaking change as users can rely on checking if the result is empty or not

Comment (Andersson007) [reply tests/integration/targets/postgresql_query/tasks/postgresql_query_initial.yml:45]: so i would suggest converting

Comment (elpavel) [reply tests/integration/targets/postgresql_query/tasks/postgresql_query_initial.yml:45]: I think both versions return list of dictionaries, when there is data to return.
The only difference in behaviour is when there is no result set.

What should be returned in query_result when there is nothing to return? Empty list or empty dictionary? 
Using Psycopg 3 we get an empty list `[]` which may be the correct option.

The options would be to 
a) fix the existing behaviour 
b) keep the existing behaviour 
c) accept existing behaviour but fix it for psycopg3.

Comment (Andersson007) [reply tests/integration/targets/postgresql_query/tasks/postgresql_query_initial.yml:45]: i think it should continue to return what it returns now when empty, i.e. `[{}]`, until there are reasonable concerns raised
my opinion:)

Comment (elpavel) [reply tests/integration/targets/postgresql_query/tasks/postgresql_query_initial.yml]: ok, lets keep the existing behaviour

**Review by Andersson007 — APPROVED** [review]
LGTM

File: tests/integration/targets/postgresql_query/tasks/postgresql_query_initial.yml [hunk 2]
```diff
@@ -234,6 +234,37 @@
...
+    postgresql_query:
+      <<: *pg_parameters
+      query: ALTER TABLE test_table ADD COLUMN foo888 int
+      encoding: 'UTF-888-bad'
+    register: result
+    ignore_errors: true
+
+  - assert:
+      that:
+      - result.failed == true
```
[the remark this answers is not in the document]
Comment (elpavel) [reply tests/integration/targets/postgresql_query/tasks/postgresql_query_initial.yml:266]: No worries, I just copy/pasted to keep the tests consistent with the rest of the file.

**Review by hunleyd — APPROVED** [review]

**Comment by hunleyd** [comment]
do additional tests go to back branches @Andersson007 ?

**Comment by Andersson007** [comment]
> do additional tests go to back branches @Andersson007 ?

i guess if it's about Psycopg 3 it's fine not to backport

**Pull request merged**